cylc/flow/workflow_files.py

```diff
@@ -1375,27 +1375,66 @@ def infer_latest_run(
     return (path, reg)
 
 
-def check_nested_run_dirs(run_dir: Union[Path, str]) -> None:
-    """Disallow nested run dirs e.g. trying to install foo/bar where foo is
-    already a valid workflow directory.
+def check_nested_dirs(
+    path: Union[Path, str], reinstall: bool = False
+) -> None:
+    """Disallow nested dirs:
+
+    - Nested installed run dirs
+    - Nested installed workflow dirs
 
     Args:
-        run_dir: Absolute workflow run directory path.
+        path: Absolute workflow run directory path or workflow install
+            directory path.
 
-    Raises WorkflowFilesError if reg dir is nested inside a run dir.
+    Raises:
+        WorkflowFilesError if reg dir is nested inside a run dir, or an
+            install dirs are nested.
     """
-    exc_msg = (
-        "Nested run directories not allowed - cannot install workflow in "
-        "'{0}' as '{1}' is already a valid run directory."
-    )
-    reg_path = Path(os.path.normpath(run_dir))
-    for parent_dir in reg_path.parents:
+    exc_msgs = {
+        'run_dir': (
+            "Nested run directories not allowed - cannot install workflow in "
+            "'{0}' as '{1}' is already a valid run directory."),
+        'install_dir': (
+            'Nested install directories not allowed - cannot install '
+            'workflow as "{0}" is already a valid workflow install directory.')
```

Comment:
`tests/f/cylc-install/00-simple.t` shouldn't be failing the way it is. Difficult to tell from the error message what is going wrong. I suggest having the single template
```
Nested {dir_type} directories not allowed - cannot install workflow
in '{dest}' as '{existing}' is already a valid {dir_type} directory.
```
to be used like
```python
exc_msg.format(
    dir_type="install", dest=path, existing=get_cylc_run_abs_path(parent_dir)
)
```